crates/rpc/src/utils/error.rs

```diff
@@ -1,31 +1,47 @@
 use warp::{
     http::StatusCode,
     reject::Rejection,
-    reply::{Reply, with_status},
+    reply::{Reply, json, with_status},
 };
 
-use crate::types::errors::ApiError;
-
+use crate::types::errors::{ApiError, ErrorMessage};
 pub async fn handle_rejection(err: Rejection) -> Result<impl Reply, Rejection> {
     if err.is_not_found() {
-        return Ok(with_status("NOT FOUND".to_string(), StatusCode::NOT_FOUND));
+        let body = ErrorMessage {
+            code: 404,
+            message: "Not Found".to_string(),
+        };
+        return Ok(with_status(json(&body), StatusCode::NOT_FOUND));
     }
 
     if let Some(api_error) = err.find::<ApiError>() {
-        let (message, code) = match api_error {
+        let (status, message) = match api_error {
             ApiError::Unauthorized => (StatusCode::UNAUTHORIZED, "Unauthorized".to_string()),
-            ApiError::NotFound(message) => (StatusCode::NOT_FOUND, message.to_string()),
-            ApiError::BadRequest(message) => (StatusCode::BAD_REQUEST, message.to_string()),
+            ApiError::NotFound(msg) => (StatusCode::NOT_FOUND, msg.clone()),
+            ApiError::BadRequest(msg) => (StatusCode::BAD_REQUEST, msg.clone()),
             ApiError::InternalError => (
                 StatusCode::INTERNAL_SERVER_ERROR,
                 "Internal Server Error".to_string(),
             ),
+            ApiError::InvalidParameter(msg) => (
+                StatusCode::BAD_REQUEST,
+                format!("Invalid state ID: {}", msg),
+            ),
+            ApiError::ValidatorNotFound(msg) => (StatusCode::BAD_REQUEST, msg.clone()),
         };
-        return Ok(with_status(code, message));
+
+        let body = ErrorMessage {
+            code: status.as_u16(),
+            message,
+        };
+
+        return Ok(with_status(json(&body), status));
     }
 
-    Ok(with_status(
-        "INTERNAL SERVER ERROR".to_string(),
-        StatusCode::INTERNAL_SERVER_ERROR,
-    ))
+    let body = ErrorMessage {
+        code: 500,
+        message: "Internal Server Error".to_string(),
+    };
+
+    Ok(with_status(json(&body), StatusCode::INTERNAL_SERVER_ERROR))
```

Comment:
```suggestion
    Ok(with_status(ErrorMessage {
        code: 500,
        message: "Internal Server Error".to_string(),
    }.json(), StatusCode::INTERNAL_SERVER_ERROR))
```

I would do something like this